### PR TITLE
[Warrior] Added Charge, updated defaults arms

### PR DESCRIPTION
**Pull request by MatthewKyle**
Added Charge and modified by blitz
Updated Gear. APL., Settings

updated tests

fixed test outputs

### CODE REVIEW

File: sim/warrior/arms/talents.go [hunk 1]
```diff
@@ -59,6 +60,7 @@ func (war *ArmsWarrior) applyTasteForBlood() {
 		ClassSpellMask: warrior.SpellMaskRend,
 		ICD:            5 * time.Second,
 		ProcChance:     procChance,
+		SpellFlags:     core.SpellFlagAPL,
```
Comment (rosenrusinov) [inline sim/warrior/arms/talents.go:63]: You can't have triggers as APL spells. APL spells are spells that the agent can cast from the APL Cast action

File: sim/warrior/arms/talents.go [hunk 2]
```diff
@@ -50,6 +50,7 @@ func (war *ArmsWarrior) applyTasteForBlood() {
 		ActionID: actionID,
 		Duration: time.Second * 9,
 		Tag:      warrior.EnableOverpowerTag,
+		Icd:      &core.Cooldown{Duration: time.Second * 6},
```
Comment (rosenrusinov) [inline sim/warrior/arms/talents.go:53]: this ICD is doing nothing atm. In the handler for activating this aura you need to check if the ICD is ready then activate and use the ICD

File: sim/warrior/charge.go [hunk 3]
```diff
@@ -0,0 +1,33 @@
+package warrior
+
+import (
+	"time"
+
+	"github.com/wowsims/cata/sim/core"
+)
+
+func (warrior *Warrior) RegisterCharge() {
```
Comment (rosenrusinov) [inline sim/warrior/charge.go:9]: Can charge be used in combat and with no minimum range now?

File: sim/warrior/charge.go [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+	metrics := warrior.NewRageMetrics(core.ActionID{SpellID: 100})
+
+	warrior.Charge = warrior.RegisterSpell(core.SpellConfig{
+		ActionID:       core.ActionID{SpellID: 100},
+		SpellSchool:    core.SpellSchoolPhysical,
+		Flags:          core.SpellFlagAPL,
+		ClassSpellMask: SpellMaskCharge,
```
Comment (rosenrusinov) [inline sim/warrior/charge.go:16]: missing ProcMask

**Review by rosenrusinov — CHANGES_REQUESTED** [review]
Cant mark non castable spells as APL

File: sim/warrior/arms/talents.go [hunk 5]
```diff
@@ -59,6 +60,7 @@ func (war *ArmsWarrior) applyTasteForBlood() {
 		ClassSpellMask: warrior.SpellMaskRend,
 		ICD:            5 * time.Second,
```
Comment (rosenrusinov) [inline sim/warrior/arms/talents.go:61]: This ICD is 5s for the proc but the application aura is 6s. This looks weird to me

File: sim/warrior/arms/talents.go (hunk 2 above)
Comment (MatthewKyle) [reply sim/warrior/arms/talents.go:53]: Yeah, this is only for tracking the auras cooldown but required the SpellAPLFlag on the Trigger to use.  Removiung till I figure something better out.

File: sim/warrior/charge.go (hunk 3 above)
Comment (MatthewKyle) [reply sim/warrior/charge.go:9]: Juggernaut makes Charge and Intercept share a CD.  I did not add intercept as there is no DPS use currently.

File: sim/warrior/arms/talents.go (hunk 1 above)
Comment (MatthewKyle) [reply sim/warrior/arms/talents.go]: Removed.  I was trying to make the proc ICD visible.

**Pull request merged**